glotaran/optimization/objective.py

```diff
@@ -17,16 +16,32 @@
 from glotaran.parameter.parameter import Parameter
 
 if TYPE_CHECKING:
+    from glotaran.model.element import ElementResult
     from glotaran.model.experiment_model import ExperimentModel
     from glotaran.typing.types import ArrayLike
 
 
+def add_svd_to_result_dataset(dataset: xr.Dataset, global_dim: str, model_dim: str):
```

Comment:
Please use `glotaran.io.prepare_dataset.add_svd_to_dataset` where the [issue that causes the CI to fail](https://github.com/glotaran/pyglotaran/actions/runs/10770808052/job/29864958225?pr=1534) was [already fixed on staging](https://github.com/glotaran/pyglotaran/commit/2ef4ceabb9a5d9f2bd8476a32dd1c10703dbf972).

Reply:
I rather like to move the function from io to here to avoid dependency of optimization to anything other than model.

I will copy the code from io with the fixes of course. This was just a quick fix to get IO out of the equation for dev.

Reply:
Please keep in mind that you can't just move the function completely (needs re import or deprecation), since [we use it down stream](https://github.com/glotaran/pyglotaran-extras/blob/ae84a0c50ed5d3368b36a95f2a0ee9f84e58cf74/pyglotaran_extras/plotting/plot_svd.py#L7).